lib/src/facts/cache.cc

```diff
@@ -0,0 +1,116 @@
+#include <internal/facts/cache.hpp>
+#include <internal/facts/external/json_resolver.hpp>
+
+#include <rapidjson/document.h>
+#include <rapidjson/prettywriter.h>
+#include <leatherman/logging/logging.hpp>
+#include <leatherman/file_util/directory.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <boost/nowide/fstream.hpp>
+#include <time.h>
+
+using namespace std;
+using namespace rapidjson;
+using namespace facter::facts::external;
+namespace boost_file = boost::filesystem;
+
+namespace facter { namespace facts { namespace cache {
+
+    bool cache_is_valid(boost_file::path const& cache_dir, int64_t ttl) {
+        time_t last_mod = boost_file::last_write_time(cache_dir);
```

Comment:
The last modified datetime stamps on directories under Windows are notoriously erroneous. Given that only puppet _should_ be modifying these it may be ok, but I'm not comfortable with the directory based timestamps.  The file based stamps are far more reliable